VisualizationBase/src/items/ViewItem.cpp

```diff
@@ -169,22 +206,43 @@ QList<QPair<Item*, Item*>> ViewItem::arrowsForLayer(QString layer)
 			auto allFrom = fromParent->findAllVisualizationsOf(line.from_);
 			auto toParent = line.toParent_ ? findVisualizationOf(line.toParent_) : this;
 			auto allTo = toParent->findAllVisualizationsOf(line.to_);
-			if (allFrom.size() > 0 && allTo.size() > 0)
-			{
-				for (auto from : allFrom)
-					for (auto to : allTo)
-						//TODO@cyril The node is rendered as a RootItem -> needs a hack here
-						//or removing a ViewItemNode with arrows doesn't work correctly
-						if (!DCast<RootItem>(from) && !DCast<RootItem>(to))
-							arrows_[line.layer_].append(QPair<Item*, Item*>(from, to));
-			}
-			else arrowsToAdd_.append(line);
+
+			bool addedArrow = false;
+			for (auto from : allFrom)
+				for (auto to : allTo)
+					//TODO@cyril The node is rendered as a RootItem -> needs a hack here
```

Comment:
Well, essentially what happens is that in the renderer we render ViewItems as RootItems if I remember correctly. So what happens here, is that there would be two items visualizing the same ViewItemNode (the RootItem and the VViewItemNode), and arrows would exist multiple times. That causes an issue when an item is then deleted.